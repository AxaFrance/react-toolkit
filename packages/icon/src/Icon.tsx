import React from 'react';
import {
  withClassDefault,
  withClassModifier,
  WithClassModifierOptions,
  compose,
} from '@axa-fr/react-toolkit-core';

const DEFAULT_CLASSNAME = 'af-icons';

interface IconBaseProps {
  icon: string;
  basePath?: string;
}

const defaultProps: Partial<IconBaseProps> = {
  basePath: '/',
};

const IconRaw: React.SFC<IconBaseProps> = ({ icon, basePath }) => (
  <svg viewBox="0 0 16 16">
    {icon && <use xlinkHref={`${basePath}icons.svg#${icon}`} />}
  </svg>
);

IconRaw.defaultProps = defaultProps;

export type IconProps = IconBaseProps & WithClassModifierOptions;

const enhance = compose<IconBaseProps, IconProps>(
  withClassDefault(DEFAULT_CLASSNAME),
  withClassModifier
);

const Enhanced = enhance(IconRaw);
Enhanced.displayName = 'Icon';
export default Enhanced;
