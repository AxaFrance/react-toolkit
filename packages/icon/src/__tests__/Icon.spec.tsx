import React from 'react';
import { render } from '@testing-library/react';
import Icon from '../Icon';

describe('<Icon>', () => {
  it('should render a sante icon', () => {
    const { asFragment } = render(<Icon icon="sante" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
