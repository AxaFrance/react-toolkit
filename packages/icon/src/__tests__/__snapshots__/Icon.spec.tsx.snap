// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon> should render a sante icon 1`] = `
<DocumentFragment>
  <svg
    viewBox="0 0 16 16"
  >
    <use
      xlink:href="/icons.svg#sante"
    />
  </svg>
</DocumentFragment>
`;
